var/spack/repos/builtin/packages/kokkos/package.py

```diff
@@ -281,3 +285,47 @@ def cmake_args(self):
         options.append('-DBUILD_SHARED_LIBS=%s' % ('+shared' in self.spec))
 
         return options
+
+    test_script_relative_path = "scripts/spack_test"
+
+    @run_after('install')
+    def setup_build_tests(self):
+        # Skip if unsupported version
+        cmake_source_path = join_path(self.stage.source_path,
+                                      self.test_script_relative_path)
+        if not os.path.exists(cmake_source_path):
+            return
+        """Copy test."""
+        cmake_out_path = join_path(self.test_script_relative_path, 'out')
+        cmake_args = [cmake_source_path,
+                      "-DSPACK_PACKAGE_SOURCE_DIR:PATH={0}".
+                      format(self.stage.source_path),
+                      "-DSPACK_PACKAGE_TEST_ROOT_DIR:PATH={0}".
+                      format(join_path(self.install_test_root, cmake_out_path)),
+                      "-DSPACK_PACKAGE_INSTALL_DIR:PATH={0}".format(self.prefix)]
+        cmake(*cmake_args)
+        self.cache_extra_test_sources(cmake_out_path)
+
+    def build_tests(self):
+        """Build test."""
```

Comment:
What is the working directory here? This might clash with build directory.

Reply:
Chiming in here: As long as this is called from the context of Spack smoke tests (i.e., from within `def test`), the directory will be within the package's (smoke) test stage directory.  (See https://spack.readthedocs.io/en/latest/packaging_guide.html?highlight=test_stage#install-tests.)

Reply:
Yes, we should be good here.